sharder/deterministic.go

```diff
@@ -122,28 +123,50 @@ func (d *DeterministicSharder) Start() error {
 		if err != nil {
 			return errors.Wrap(err, "failed to get local interface list to initialize sharder")
 		}
+		localAddrCidrs := make([]string, len(localAddrs))
+		for i, addr := range localAddrs {
+			localAddrCidrs[i] = addr.String()
+		}
+
+		// If RedisIdentifier is an IP, add it to localAddrs.
```

Comment:
Does it actually make sense to add it to `localCidrAddrs` (typo there), or would it be better to simply replace the contents of `localCidrAddrs` with a single value -- the one specified by a non-empty `GetRedisIdentifier()`?

Reply:
Fixed.

Reply:
Thanks -- I should have split this into two questions. If we use RedisIdentifier, I don't think we need to search the entire list of IPs we get by querying the interface. We should *only* find RedisIdentifier anyway -- so instead of appending it to the list, we could replace the whole thing with a list containing one element.

Or am I missing something?

Reply:
I took the minimal code change approach to accomplish the desired functionality.  I admit not understanding why this self lookup is necessary.  Rather than break whatever those use cases are, I just expanded on it.

If it is decided to replace the list, then I'll make that change.  Just please confirm.